### PR TITLE
Overhaul README: Wording etc.

### PR DESCRIPTION
**Pull request by Olf0**

### CODE REVIEW

**Comment by Olf0** [comment]
@rinigus, if you have questions, suggestions, requests for adaptions etc., please let us discuss them first here.
You as the principal maintainer of SailfishOS:Chum implicitly have the final say ("last word") on anything happening in this repo, but OTOH you know by now, that I set (almost) every character in a text deliberately.  While switching off "maintainer commits" would be childish (and futile), I ask you not to use them without discussing first.

**Comment by rinigus** [comment]
Started reading it and would be happy to discuss. OK, this is then the first PR we have to handle, before #49 . Much of docs has been handled by @piggz, pinging him as well :)

File: README.md [hunk 1]
```diff
@@ -1,197 +1,152 @@
...
-  be built and shipped on OBS.  There may be exceptions like some firmware 
-  blobs but in general, follow the guiddance at:
-  https://en.opensuse.org/openSUSE:Build_Service_application_blacklist
+- Can I use the RPMs of my software built at SailfishOS:Chum to upload them to the Jolla Store?<br />
+  No, you cannot, because RPMs built at SailfishOS:Chum have the vendor set to `chum`, which is not allowed at the Jolla Store ("harbour").
+  However, it is easy to set up a personal repository at the Sailfish&nbsp;OS OBS, configure `sailfishos:chum` to provide the required dependencies and re-build your packages at your own repository.
+  As a result, you will get automated builds for all architectures wanted without the vendor set to `chum` in your RPMs.
+
+- Can I use the RPMs of my software built at SailfishOS:Chum to upload them to OpenRepos or elsewhere?<br />
+  While you could do that, it is not recommended to re-distribute RPMs from SailfishOS:Chum because they all have the vendor set to `chum`, which will prevent users from distinguishing whether a package was directly installed from the SailfishOS:Chum repository or from some other package repository.<br />
```
Comment (rinigus) [inline README.md:146]: Same linebreak :)

**Review by rinigus — COMMENTED** [review]
Thank you for revising the README. All in all, I think text has improved. 

Almost all suggested changes are removal of forced HTML linkebreaks. You are also using nbsp; in the text. Which is understandable (we don't seem to have `~` in markdown), but it's usage will slip in future. Personally, I don't mind linebreaks between Sailfish and OS, but it is OK to use your notation.

File: README.md [hunk 2]
```diff
@@ -1,197 +1,152 @@
...
-By collecting the software in a single automated build system, we can
-benefit from collaboration between developers through shared packaging
-of required libraries, reduce duplication of the work by keeping the
-packages up to date, and get a clear overview of missing software.
-
-The repository is located at OBS:
-- https://build.sailfishos.org/project/show/sailfishos:chum
+# SailfishOS:Chum community repository: Documentation and issue tracker
+
+The SailfishOS:Chum community repository is providing collection of
```
Comment (piggz) [inline README.md:3]: is providing -> provides a

File: README.md [hunk 3]
```diff
@@ -1,197 +1,152 @@
...
-```
+### How to deploy the configuration for command line tools
 
+For using the SailfishOS:Chum repository per command line tools, a `sailfishos-chum` helper RPM is available for easy installation also at [chumrpm.netlify.app](https://chumrpm.netlify.app/) (and additionally at the [SailfishOS:Chum repository](https://build.sailfishos.org/package/show/sailfishos:chum/sailfishos-chum) proper), which solely provides an appropriate local repository configuration for utilising the SailfishOS:Chum repository.<br />
+Note that installing the SailfishOS:Chum GUI-client deploys the same local repository configuration, so your device is already set for using the SailfishOS:Chum repository with the usual command line tools for package management, then.
 
-## How to use these repositories: developers
+To utilise the SailfishOS:Chum repository per command line tools on your device, you have to …
+1. download either of the aforementioned packages `sailfishos-chum` or `sailfishos-chum-gui`.
+2. install it per file manager or as root user per `pkcon install-local sailfishos-chum[-gui]` or `zypper in sailfishos-chum[-gui]`.
```
[the remark this answers is not in the document]
Comment (Olf0) [reply README.md:61]: True something follows, which is not denoted, but wouldn't `sailfishos-chum[-gui]-<version>.rpm` indicate the true situation better?

File: README.md [hunk 4]
```diff
@@ -1,197 +1,152 @@
...
+3. refresh the software cache on your device as root user per `pkcon refresh` or `zypper ref`.
+4. install any software packages you like from the SailfishOS:Chum repository as root user per `pkcon install <package name>` or `zypper in <package name>`.
 
-There are no major limitations imposed on the submitted software. Main
-limitation is that it should not interfere with any software
-distributed by Jolla as a part of Sailfish OS. This includes updating
-libraries distributed as a part of Sailfish OS.
+As all software packages at SailfishOS:Chum have their vendor set to `chum`, it is easy to determine which packages are installed from the SailfishOS:Chum repository by executing:<br />
+`rpm -qa --queryformat '%{vendor}:%{name}\n' | fgrep chum`
```
[the remark this answers is not in the document]
Comment (Olf0) [reply README.md:66]: Seriously, for a single line of pre-formatted text?
That renders as a box with pre-formatted text, which is usually only used for multiple lines of pre-formatted text (i.e., a `<pre>` text block), consuming a lot of space unnecessarily.

File: README.md (hunk 2 above)
Comment (Olf0) [reply README.md:3]: Ack.

File: README.md [hunk 5]
```diff
@@ -1,197 +1,152 @@
...
+packages are generated solely from this source code. Hence all software
+packages at SailfishOS:Chum are created in a transparent and fully
+traceable manner.
+
+By collecting software for Sailfish&nbsp;OS in a single automated build
+system, collaboration between developers through common packaging of
+shared libraries etc. is fostered, duplication of work for keeping these
+common packages up-to-date is eliminated, and it becomes much easier to
+determine which pieces of software exist and which are missing at the
+Sailfish&nbsp;OS OBS. Additionally this eases to trace multiple and
```
Comment (piggz) [inline README.md:29]: to trace -> tracing

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rinigus) [reply README.md:61]: yes, that's better

File: README.md [hunk 6]
```diff
@@ -1,197 +1,152 @@
...
-```sh
-rpm -qa --queryformat '%{vendor}:%{name}\n' | grep chum
-```
+### How to deploy the configuration for command line tools
 
+For using the SailfishOS:Chum repository per command line tools, a `sailfishos-chum` helper RPM is available for easy installation also at [chumrpm.netlify.app](https://chumrpm.netlify.app/) (and additionally at the [SailfishOS:Chum repository](https://build.sailfishos.org/package/show/sailfishos:chum/sailfishos-chum) proper), which solely provides an appropriate local repository configuration for utilising the SailfishOS:Chum repository.<br />
+Note that installing the SailfishOS:Chum GUI-client deploys the same local repository configuration, so your device is already set for using the SailfishOS:Chum repository with the usual command line tools for package management, then.
 
-## How to use these repositories: developers
+To utilise the SailfishOS:Chum repository per command line tools on your device, you have to …
```
Comment (piggz) [inline README.md:59]: remove "per command line tools"

File: README.md (hunk 3 above)
Comment (piggz) [inline README.md:61]: per -> using the

File: README.md [hunk 7]
```diff
@@ -1,197 +1,152 @@
...
+1. download either of the aforementioned packages `sailfishos-chum` or `sailfishos-chum-gui`.
+2. install it per file manager or as root user per `pkcon install-local sailfishos-chum[-gui]` or `zypper in sailfishos-chum[-gui]`.
+3. refresh the software cache on your device as root user per `pkcon refresh` or `zypper ref`.
```
Comment (piggz) [inline README.md:62]: per -> using

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (rinigus) [reply README.md:66]: well - feel free to ignore it and mark as "resolved". I would have done it using ```

**Comment by piggz** [comment]
One last change I would make ... add a link and description of adding the chum store meta data to the rpm .spec file

**Comment by Olf0** [comment]
> Thank you for revising the README. All in all, I think text has improved.

Good. 

> Almost all suggested changes are removal of forced HTML linkebreaks.

I see, though I tried to use them sparingly.

IMO they are less awkward than the strange tricks used before:
- Using a single line unordered list to achieve a forced line break.
- Using a single line pre-formatted text block to achieve a forced line-break.

Furthermore, the looks are really awkward at some other places without forced line breaks:
- The FAQ questions look really strange, if not solitary on a line, i.e., if the answer is directly appended to it.
  But an empty line breaks a list element into a list element and an independent line outside of the list.
- URLs can span over a line end (at least I saw that in the preview): It makes them harder to see, click and comprehend that it is not two URLs, if they are not within a single line.
- Sometimes it just makes sense to emphasise a sentence within a paragraph by letting it begin at the start of a line.  Usually because it is important and shall not be quickly read over.  Breaking the paragraph into two breaks the logical cohesion between the sentences, hence this is not a "solution", because it significantly alters the structure and meaning of a text.

In know well, that one should not try to force a HTML renderer into one's own ideas of formatting (because the easily backfires), and this is not what I want to achieve here.  But your apparently strict policy of never using a \<br /\> is hard to comprehend for me in above cases, especially when you suggest workarounds which definitely fall into the category "but that is not how it was intended to be used" (technically in the first two examples, semantically the other three).
Can you please briefly denote, where your strong aversion of line-breaks originates.

At least I now understand why you repeatedly felt the urge to either eliminate the single line-breaks or convert them into a double line break when submitting my suggestions from issue #45.

BTW (& JFYI), I eliminated two forced line-breaks in PR #50: I avoid them when easily possible.

> You are also using nbsp; in the text. 

Which I hate to use, but IMO it is less evil than a line-break between "Sailfish" and "OS".
I wish Jolla would have never altered the spelling from "SailfishOS" (before 2015) to "Sailfish&nbsp;OS".  I usually ignore this change, but your "ambition is to become the principal software distribution for Sailfish&nbsp;OS".  So I expected some appeasement to Jolla's spelling rules to be necessary.

File: README.md (hunk 7 above)
Comment (Olf0) [reply README.md:62]: Ack.

File: README.md (hunk 3 above)
Comment (Olf0) [reply README.md:61]: O.K.

File: README.md (hunk 5 above)
Comment (Olf0) [reply README.md:29]: Yes.

File: README.md (hunk 6 above)
Comment (Olf0) [reply README.md:59]: Superseded by the next comment, I suppose.

File: README.md (hunk 7 above)
Comment (Olf0) [reply README.md:63]: … decided for "by executing" the second occurrence.

**Comment by Olf0** [comment]
> One last change I would make ... add a link and description of adding the chum store meta data to the rpm .spec file

Done.

**Comment by rinigus** [comment]
> * The FAQ questions look really strange, if not solitary on a line, i.e., if the answer is directly appended to it.
>       But an empty line breaks a list element into a list element and an independent line outside of the list.

Notice that this is a location where I haven't suggested to drop breaks.



> * URLs can span over a line end (at least I saw that in the preview): It makes them harder to see, click and comprehend that it is not two URLs, if they are not within a single line.

In practice, you just click at URL and focus on the next text rendered after clicking on it. I wouldn't worry about it - although it could have been easily my own hacky "workaround". 

> * Sometimes it just makes sense to emphasise a sentence within a paragraph by letting it begin at the start of a line.  Usually because it is important and shall not be quickly read over.  Breaking the paragraph into two breaks the logical cohesion between the sentences, hence this is not a "solution", because it significantly alters the structure and meaning of a text.

Here I would oppose it. If it is within the paragraph then it should just be in there. If there is something important in the middle of the paragraph that you don't want to miss it then we have to either rewrite the paragraph and add that message in front or split the paragraphs into two. Having a linebreak within the paragraph, in my opinion, is not acceptable when we have a regular text. We can look through many texts from different backgrounds and it is rather common approach throughout.

So, please, if you wish to use a linebreak, unless we have such specific case as FAQ, then either decide whether it is new paragraph or skip it.

I guess the text above responds to your more general question regarding linebreaks. 

> Which I hate to use, but IMO it is less evil than a line-break between "Sailfish" and "OS".

I think this is OK as MarkDown is missing such feature as non-breakable-space, as we have in few other places. Just we will probably forget to add it and happily let it break there.

**Comment by rinigus** [comment]
Thank you very much for the changes! LGTM

**Comment by Olf0** [comment]
@rinigus,
> > * The FAQ questions look really strange, if not solitary on a line, i.e., if the answer is directly appended to it.
> >   But an empty line breaks a list element into a list element and an independent line outside of the list.
> 
> Notice that this is a location where I haven't suggested to drop breaks.

Well, https://github.com/sailfishos-chum/main/pull/48#discussion_r785462090.
Thus they are gone now.

> > * URLs can span over a line end (at least I saw that in the preview): It makes them harder to see, click and comprehend that it is not two URLs, if they are not within a single line.
> 
> In practice, you just click at URL and focus on the next text rendered after clicking on it. I wouldn't worry about it - although it could have been easily my own hacky "workaround".

See https://github.com/sailfishos-chum/main/blame/72f41cb65626ad16255d6a7d0863496ca7762036/README.md#L23.
Later only the URL was adapted by flypig from `build.merproject.org` to `build.sailfishos.org`, until https://github.com/sailfishos-chum/main/pull/48#discussion_r785457642.
But the "hacky workaround" and my forced line-break are gone now.
 
> > * Sometimes it just makes sense to emphasise a sentence within a paragraph by letting it begin at the start of a line.  Usually because it is important and shall not be quickly read over.  Breaking the paragraph into two breaks the logical cohesion between the sentences, hence this is not a "solution", because it significantly alters the structure and meaning of a text.
> 
> Here I would oppose it. If it is within the paragraph then it should just be in there. If there is something important in the middle of the paragraph that you don't want to miss it then we have to either rewrite the paragraph and add that message in front or split the paragraphs into two. Having a linebreak within the paragraph, in my opinion, is not acceptable when we have a regular text. We can look through many texts from different backgrounds and it is rather common approach throughout.

To me this is a style question and there also are many texts handling this "my way", thus I am still surprised that you think of "*Having a linebreak within the paragraph, in my opinion, is not acceptable when we have a regular text.*", because "*not acceptable*" is a very strong rsp. strict stance.
Anyway, we gained some mutual understanding, which is a good outcome from my POV.

**Comment by Olf0** [comment]
Looking at the README as rendered at [the sailfishos-chum/main frontpage](https://github.com/sailfishos-chum/main#sailfishoschum-community-repository-documentation-and-issue-tracker) now, made me wish to re-insert forced line-breaks at least at the locations where @rinigus indicated that they are acceptable for him.
Thus when I spotted a typo, I took the chance: PR #51

**Comment by rinigus** [comment]
@Olf0: agree about strong wording - should have used some lighter tone. Sorry about it.

**Comment by Olf0** [comment]
> @Olf0: agree about strong wording - should have used some lighter tone. Sorry about it.

Socially that was O.K. from my perspective, thus no need to say "sorry": You expressed your opinion WRT "forced line breaks" very clearly, but it never was directed "ad hominem".
And I always prefer unambiguous statements over soft, ambiguous ones: "Clarity is paramount!"

Hence I do not think you "*should have used some lighter tone*", if that was your opinion at that time.
But I am very glad that you seem to have "lightened" the strength of your opinion a bit, because in my understanding your original stance left no room to apply "forced line breaks" at all, which conflicts with my belief that they are sometimes (although rarely) necessary.

Thus gaining some mutual understanding and seeing that PR #51 apparently was O.K. for you is a proper base for future contributions from my perspective.
Thanks!

P.S.: Actually I have one file on my radar, which contains "forced line breaks" at places where even I do not think they are appropriate.  A PR from me for fixing that might come some time.

**Comment by rinigus** [comment]
I am sure we can strike a balance, as we did with PR #51. Would just have to be sure that the break is, indeed, very needed there.

**Pull request merged**